gammapy/time/plot.py

```diff
@@ -30,3 +28,140 @@ def plot_time_difference_distribution(time, ax=None):
 
     # TODO: implement!
     raise NotImplementedError
+
+def plot_light_curve(name_3FGL, time_start, time_end, ax = None):
+    """Plot flux as a function of time for a fermi 3FGL object.
+
+    Parameters
+    ----------
+    name_3FGL : `string`
+        The 3FGL catalog name of the object to plot
+    fermi_met_start : `~astropy.time.Time`
+        Astropy time object for the start of the light curve
+    fermi_met_end : `~astropy.time.Time`
+        Astropy time object for the end of the light curve
+    ax : `~matplotlib.axes.Axes` or None
+        Axes
+    Returns
+    -------
+    ax : `~matplotlib.axes.Axes`
+
+    Examples
+    --------
+    Plot effective area vs. energy:
+
+    .. plot::
+        :include-source:
+
+        import gammapy.time
+        import astropy.time
+        import matplotlib.pyplot as plt
+
+        time_start = astropy.time.Time('2010-01-01T00:00:00')
+        time_end = astropy.time.Time('2015-02-02T02:02:02')
+
+        plt.plot = gammapy.time.plot_light_curve('3FGL J0349.9-2102', time_start, time_end)
+        plt.show()
+    """
+    from ..datasets import fetch_fermi_catalog
+    import astropy.time
+    import numpy as np
+    import matplotlib.pyplot as plt
+    import matplotlib.dates as mdates
+    import math
+
+    ax = plt.gca() if ax is None else ax
+
+    fermi_met_base = astropy.time.Time('2001-01-01T00:00:00')
+
+    fermi_met_start = (time_start - fermi_met_base).sec
+
+    fermi_met_end = (time_end - fermi_met_base).sec
+
+    # As far as I know light curves were only included in 3FGL, so we must use this catalog.
+    fermi_cat = fetch_fermi_catalog('3FGL')
+
+    catalog_index = np.where(fermi_cat[1].data['Source_Name'] == name_3FGL)[0][0]
+
+    time_index_start = np.where(fermi_cat[3].data['Hist_Start'] >= fermi_met_start)[0][0]
+
+    # The final entry is the end of the last bin, so no off by one error
+    time_index_end = np.where(fermi_cat[3].data['Hist_Start'] <= fermi_met_end)[0][-1] + 1
+
+    time_start = fermi_cat[3].data['Hist_Start'][time_index_start: time_index_end]
+    time_end = np.roll(time_start, -1)
+
+    time_diff = time_end - time_start
+
+    # Trim because there is one more bin edge than there is bin mid point
+    time_diff = time_diff[0:-1]
+
+    # Midpoints of each bin.
+    time_mid = time_start[0:-1] + time_diff
+
+    flux_history = fermi_cat[1].data[catalog_index]['Flux_History'][time_index_start: time_index_end]
+
+    flux_history_lower_bound = fermi_cat[1].data[catalog_index]['Unc_Flux_History'][time_index_start: time_index_end, 0]
```

Comment:
Such long lines are hard to read. In this case, please extract `fermi_cat[1].data[catalog_index]` out in a temp variable to shorten the line and avoid duplicated code.